src/main/scala/com.snowplowanalytics/sqs2kinesis/EventsStreamModule.scala

```diff
@@ -58,89 +55,133 @@ object EventsStreamModule {
       client
     }
 
-    val KinesisKey = "kinesisKey"
+    val decider: Supervision.Decider = { e =>
+      logger.error("Error in stream", e)
+      Supervision.Resume
+    }
+
+    sqsSource(config)
+      .via(sqsMsg2kinesisMsg)
+      .via(Batcher.batch(
+        keepAlive          = 1.second,
+        maxSize            = MaxKinesisBatch,
+        maxWeight          = MaxKinesisBytesPerRequest,
+        toWeight           = {m => m.bytes.size + m.key.getBytes.size}
+      ))
+      .via(kinesisFlow(config, kinesisClient))
+      .alsoTo(confirmSqsSink(config))
+      .alsoTo(printProgress)
+      .toMat(Sink.ignore)(Keep.right)
+      .withAttributes(ActorAttributes.supervisionStrategy(decider))
+      .run()
+  }
+
+  val KinesisKey = "kinesisKey"
 
-    val sqsSource: Source[Message, NotUsed] =
+  /** A source that reads messages from sqs and retries if read was unsuccessful */
+  def sqsSource(config: Sqs2KinesisConfig)(implicit client: SqsAsyncClient): Source[Message, NotUsed] =
+    RestartSource.withBackoff(500.millis, 1.second, 0.1) { () =>
       SqsSource(
         config.sqsQueue,
         SqsSourceSettings.Defaults.withMessageAttribute(MessageAttributeName(KinesisKey))
       )
+    }
 
-    type KinesisKeyAndMsg = (String, ByteBuffer)
+  case class ParsedMsg(original: Message, key: String, bytes: Array[Byte])
 
-    val sqsMsg2kinesisMsg: Message => KinesisKeyAndMsg =
-      msg => {
-        val decoded  = ByteBuffer.wrap(java.util.Base64.getDecoder.decode(msg.body))
+  /** A flow that base64-decodes sqs messages. It ignores messages if they cannot be parsed */
+  val sqsMsg2kinesisMsg: Flow[Message, ParsedMsg, NotUsed] =
+    Flow[Message]
+      .map { msg =>
+        logger.debug(s"Received message ${msg.messageId}")
+        val decoded  = java.util.Base64.getDecoder.decode(msg.body)
         val maybeKey = Option(msg.messageAttributes().get(KinesisKey)).map(_.stringValue())
         val key = maybeKey.getOrElse {
           val randomKey = UUID.randomUUID().toString
           logger.warn(s"Kinesis key for sqs message ${msg.messageId()} not found, random key generated: $randomKey")
           randomKey
         }
-        (key, decoded)
+        Some(ParsedMsg(msg, key, decoded))
+      }.recover {
+        case NonFatal(e) =>
+          logger.error("Error decoding sqs message. Message will be ignored and not acked.", e)
```

Comment:
That would certainly be convenient.  In most other places in our pipeline I think we don't log events to stdout/stderr, and I assumed this is because of concerns about pii.  I wonder if we have a rule or guidance about this?

Here, the offending message is written to the bad topic.  Hopefully this would help with debugging to some extent, even if it's not as convenient as having it written directly to the log.